### DISCUSSION TITLE
Property with the SeralizedPath-Attribute is not properly converted by Serializer

### DISCUSSION THREAD

**Discussion by Nico-Richter-TSC** [symfony/symfony#49225] [Q&A] [answered]
Hi!

I have the problem that a DTO with `SerializedPath` attributes, which I convert to an XML via Symfony Serializer (`SerializerInterface`), does not have an expected behavior with arrays. Unfortunately, the documentation for this is small due to the novelty (`SerializedPath` is added in 6.2), so I can't get anywhere without help. 

My goal is to create an XML like this one:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
    <methods>
        <include code="Method A"/>
        <include code="Method B"/>
    </methods>
</root>
```
For example with such a class:
```php
class TestClass
{
    #[SerializedPath('[methods][include][@code]')]
    public array $methods = ['Method A', 'Method B'];
}

// ...

$this->serializer->serialize(new TestClass(), 'xml');
```

But I just get always the same issue:
```json
{
    "title": "An error occurred",
    "detail": "The element you are trying to set is already populated: \"[methods][include][@code]\".",
    "trace": [
        {
            "file": "/srv/api/vendor/symfony/serializer/Normalizer/AbstractObjectNormalizer.php",
            "line": 655
        }
   ]
}
```

I already looked at the documentation for the `SerializedPath` (https://symfony.com/doc/current/serializer.html#using-nested-attributes) and also for the `PropertyAccess`which is internal used (https://symfony.com/doc/current/components/property_access.html#reading-from-arrays), but I don't see a suitable example from which I can work it out.

Perhaps one approaches the matter in a completely different way. I just want to avoid all the nested DTOs with SerializedName attributes.

If anyone can help, I would be very grateful!

**Comment by boenner** [comment, marked as answer]
Sounds like this is related to https://github.com/symfony/symfony/issues/49494. I've tested your code with the fix from the branch and it seems to work when changing `$methods` to a string (assigning an array to a XML attribute with the `@` prefix throws an error).
